pulsar-function-go/pf/instance.go

```diff
@@ -259,6 +268,39 @@ func getIdleTimeout(timeoutMilliSecond time.Duration) time.Duration {
 	return timeoutMilliSecond
 }
 
+func (gi *goInstance) setupLogHandler() error {
+	if gi.context.instanceConf.funcDetails.GetLogTopic() != "" {
+		gi.context.logAppender = NewLogAppender(
+			gi.client,                                         //pulsar client
+			gi.context.instanceConf.funcDetails.GetLogTopic(), //log topic
+			getDefaultSubscriptionName(gi.context.instanceConf.funcDetails.Tenant, //fqn
+				gi.context.instanceConf.funcDetails.Namespace,
+				gi.context.instanceConf.funcDetails.Name),
+		)
+		return gi.context.logAppender.Start()
+	}
+	return nil
+}
+
+func (gi *goInstance) addLogTopicHandler() {
+	if gi.context.logAppender == nil {
+		panic("please init logAppender")
+	}
+
+	for _, logByte := range log.StrEntry {
```

Comment:
OK, there is a bad place in this place. Regarding the processing logic of StrEntry, I should put this pull request, so the whole code logic may be relatively clear.

The specific processing logic is as follows:

```
var StrEntry []string

func (hook *contextHook) Fire(entry *log.Entry) error {
        ...
        //handler log appender
	strEntry, err := entry.String()
	if err != nil {
		fmt.Fprintf(os.Stderr, "Unable to read entry, %v", err)
		return err
	}
	StrEntry = append(StrEntry, strEntry)
	return nil
}
```

In go function, we used used is [logrus](https://github.com/sirupsen/logrus) as log lib. He has a powerful function **hook**. Users can implement hooks themselves to ensure that each time the log is written, the Fire() method in the hook is called. With this feature,  i put the logs I need to write every time into []StrEntry, so that in our appender, we can get the data we want from this []StrEntry, and then send it to the producer's msg payload for corresponding processing.